### PR TITLE
UCT/IB: Fallback to ibv_create_cq() if ibv_create_cq_ex() returns ENOSYS - v1.5.x

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -636,11 +636,17 @@ struct ibv_cq *uct_ib_create_cq(struct ibv_context *context, int cqe,
...
+    if (cq_ex) {
+        return ibv_cq_ex_to_cq(cq_ex);
+    } else if (errno != ENOSYS) {
+        return NULL;
+    } else
+        /* if ibv_create_cq_ex returns ENOSYS, fallback to ibv_create_cq */
 #endif
-    return cq;
+    {
+        return ibv_create_cq(context, cqe, NULL, channel, comp_vector);
```
Comment (hoopoepg) [inline src/uct/ib/base/ib_iface.c]: minor: why not just leave this line without ```{ }```? as I can see there is return from function if cq_ex is completed successfully or failed due to error

Comment (hoopoepg) [reply src/uct/ib/base/ib_iface.c]: like
```
    if (cq_ex) {
        return ibv_cq_ex_to_cq(cq_ex);
    } else if (errno != ENOSYS) {
        return NULL;
    }
    /* if ibv_create_cq_ex returns ENOSYS, fallback to ibv_create_cq */
#endif
    return ibv_create_cq(context, cqe, NULL, channel, comp_vector);
```

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10607/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10608/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10612/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**